### PR TITLE
add warning for system variable tidb_skip_utf8_check

### PR DESCRIPTION
**Pull request by TomShawn**
### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [ ] v6.2 (TiDB 6.2 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [x] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/9847 https://github.com/pingcap/docs-cn/pull/9968
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: character-set-and-collation.md [hunk 1]
```diff
@@ -376,6 +376,10 @@ If the specified character set is `utf8` or `utf8mb4`, TiDB only supports the va
 
 To disable this error reporting, use `set @@tidb_skip_utf8_check=1;` to skip the character check.
 
+> **Note:**
+>
+> If the character check is skipped, TiDB might fail to detect illegal UTF-8 characters written by the application, cause decoding errors when `ANALYZE` is executed, and introduce other unknown encoding issues. If your application cannot guarantee the validity of the written string, it is not recommended to skip the character check.
```
Comment (Oreoxmt) [inline character-set-and-collation.md:381]:
```suggestion
> If the character check is skipped, TiDB might fail to detect illegal UTF-8 characters written by the application, causing decode errors when `ANALYZE` is executed, and introduce other unknown encoding issues. If your application cannot guarantee the validity of the written string, it is not recommended to skip the character check.
```

File: system-variables.md [hunk 2]
```diff
@@ -2000,6 +2000,10 @@ Query OK, 0 rows affected, 1 warning (0.00 sec)
 - This variable is used to set whether to skip UTF-8 validation.
 - Validating UTF-8 characters affects the performance. When you are sure that the input characters are valid UTF-8 characters, you can set the variable value to `ON`.
 
+> **Note:**
+>
+> If the character check is skipped, TiDB might fail to detect illegal UTF-8 characters written by the application, cause decoding errors when `ANALYZE` is executed, and introduce other unknown encoding issues. If your application cannot guarantee the validity of the written string, it is not recommended to skip the character check.
```
Comment (Oreoxmt) [inline system-variables.md:2005]:
```suggestion
> If the character check is skipped, TiDB might fail to detect illegal UTF-8 characters written by the application, causing decode errors when `ANALYZE` is executed, and introduce other unknown encoding issues. If your application cannot guarantee the validity of the written string, it is not recommended to skip the character check.
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

File: character-set-and-collation.md (hunk 1 above)
Comment (TomShawn) [reply character-set-and-collation.md:381]: might [fail to, cause, and introduce]

**Pull request merged**